test/GPv2Settlement.e2e.ts

```diff
@@ -0,0 +1,201 @@
+import ERC20 from "@openzeppelin/contracts/build/contracts/ERC20PresetMinterPauser.json";
+import { expect } from "chai";
+import { BigNumber, Contract, TypedDataDomain, Wallet } from "ethers";
+import { ethers, waffle } from "hardhat";
+
+import {
+  Order,
+  OrderKind,
+  SettlementEncoder,
+  SigningScheme,
+  allowanceManagerAddress,
+  domain,
+} from "../src/ts";
+
+describe("GPv2Settlement: End to End Tests", () => {
+  const [deployer, owner, solver, ...traders] = waffle.provider.getWallets();
+
+  let settlement: Contract;
+  let allowanceManager: Contract;
+  let domainSeparator: TypedDataDomain;
+
+  beforeEach(async () => {
+    const GPv2AllowListAuthentication = await ethers.getContractFactory(
+      "GPv2AllowListAuthentication",
+      deployer,
+    );
+    const authenticator = await GPv2AllowListAuthentication.deploy(
+      owner.address,
+    );
+    await authenticator.connect(owner).addSolver(solver.address);
+
+    const GPv2Settlement = await ethers.getContractFactory(
+      "GPv2Settlement",
+      deployer,
+    );
+    settlement = await GPv2Settlement.deploy(authenticator.address);
+    allowanceManager = await ethers.getContractAt(
+      "GPv2AllowanceManager",
+      allowanceManagerAddress(settlement.address),
+    );
+
+    const { chainId } = await ethers.provider.getNetwork();
+    domainSeparator = domain(chainId, settlement.address);
+  });
+
+  it("should settle red wine and olive oil market", async () => {
+    const STARTING_BALANCE = ethers.utils.parseEther("1000.0");
+    const erc20 = (symbol: string) =>
+      waffle.deployContract(deployer, ERC20, [symbol, 18]);
+
+    const eur = await erc20("EUR");
+    const oil = await erc20("OIL");
+    const wine = await erc20("RED");
```

Comment:
Just for fun, would emojis work as ERC 20 symbols?

Reply:
Going to try this

Reply:
Definitely works!:
```ts
    const eur = await erc20("€");
    const oil = await erc20("🫒");
    const wine = await erc20("🍷");
```

Reply:
Look at this masterpiece:
![image](https://user-images.githubusercontent.com/4210206/101659842-c5778c00-3a46-11eb-9fdf-16974dfeb103.png)